test/langtools/tools/javac/8278078/InvalidThisAndSuperInConstructorArgTest.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * @test /nodynamiccopyright/
+ * @bug 8278078
+ * @summary error: cannot reference super before supertype constructor has been called
+ * @compile/fail/ref=InvalidThisAndSuperInConstructorArgTest.out -XDrawDiagnostics InvalidThisAndSuperInConstructorArgTest.java
+ */
+public class InvalidThisAndSuperInConstructorArgTest  {
+
+    interface InterfaceWithDefault {
+        default String get() {
+            return "";
+        }
+    }
+
+    InvalidThisAndSuperInConstructorArgTest(String s) {
+    }
+
+    class InnerClass extends AssertionError implements InterfaceWithDefault {
+        InnerClass() {
+            super(InnerClass.super.toString());
+        }
+        InnerClass(int i) {
```

Comment:
Should we write a similar test where, instead of InnerClass.super.toString(), we pass InnerClass.super::toString (e.g. a method ref) ? I think it would be useful that all these would fail, even in method reference mode.